<!DOCTYPE html>
<html lang="en"><head>
<script src="slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="slides_files/libs/quarto-html/popper.min.js"></script>
<script src="slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/quarto-syntax-highlighting-66ab7fd5e73b7f0a764e0d49b3e29ab1.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.13">

  <meta name="author" content="Hanbin Lee">
  <title>Modelling complex traits with ancestral recombination graphs</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/theme/quarto-bbe7401fe57d4b791b917637bb662036.css">
  <link rel="stylesheet" href="style.css">
  <link href="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Modelling complex traits with ancestral recombination graphs</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Hanbin Lee 
</div>
<div class="quarto-title-author-email">
<a href="mailto:hblee@umich.edu">hblee@umich.edu</a>
</div>
        <p class="quarto-title-affiliation">
            University of Michigan, Ann Arbor
          </p>
    </div>
</div>

  <p class="date">Mar 7, 2025</p>
</section>
<section>
<section id="overview" class="title-slide slide level1 center">
<h1>Overview</h1>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p>The ancestral recombination graph (ARG) describes the evolutionary relationship between</p>
<p>genetic materials in the presence of recombination and drift</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/full_arg-0.svg" height="600"></p>
<figcaption>From <span class="citation" data-cites="Wong2024">(<a href="#/references" role="doc-biblioref" onclick="">Wong et al. 2024</a>)</span></figcaption>
</figure>
</div>
</div>
<div class="fragment fade-in-then-out">
<p>The ancestral recombination graph (ARG) describes the evolutionary relationship between</p>
<p>genetic materials in the presence of recombination and drift</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/full_arg-1.svg" height="600"></p>
<figcaption>From <span class="citation" data-cites="Wong2024">(<a href="#/references" role="doc-biblioref" onclick="">Wong et al. 2024</a>)</span></figcaption>
</figure>
</div>
</div>
<div class="fragment fade-in-then-out">
<p>The full probabilistic process is complicated</p>
<p>In this work, we condition on the realized ARG, resulting a sequence of local trees</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/local_trees.svg" height="500"></p>
<figcaption>From <a href="https://tskit.dev/tutorials/what_is.html">tskit docs</a></figcaption>
</figure>
</div>
</div>
<div class="fragment">
<p>What is the conditional distribution of a trait given the trees?</p>
<p>Since the genealogy is fixed, the only randomness that remains is mutation</p>
<p><span class="math display">\[
\text{Trait} \mid \text{Local trees} \quad \sim \quad ?
\]</span></p>
</div>
</div>
</section>
<section id="linear-mixed-model" class="slide level2">
<h2>Linear mixed model</h2>
<div class="fragment">
<p>Linear mixed models are popular in quantitative genetics <span class="math display">\[
\mathbf{y} = \underbrace{\mathbf{Z}\mathbf{u}}_{\text{random effects}} + \underbrace{\mathbf{X}\mathbf{b}}_{\text{fixed effects}} + \boldsymbol{\varepsilon}
\]</span> where <span class="math inline">\(\mathbf{Z}\)</span> includes genotyped variants and <span class="math inline">\(\mathbf{X}\)</span> is the covariate matrix</p>
<p>In particular, the SNP effects <span class="math inline">\(\mathbf{u} \sim p(\cdot)\)</span> is <em>random</em></p>
</div>
<div class="fragment">
<p>Some questions …</p>
<div>
<ul>
<li class="fragment">What’s the source of <span class="math inline">\(\mathbf{u}\)</span>’s randomness?</li>
<li class="fragment">Why are <span class="math inline">\(\mathbf{u}\)</span>’s (vector of random effects) entries independent?</li>
</ul>
</div>
</div>
<div class="fragment">
<p>We answer these questions from a genealogical perspective</p>
</div>
</section></section>
<section>
<section id="setup-and-derivation" class="title-slide slide level1 center">
<h1>Setup and derivation</h1>
<div class="fragment">
<p>The trait <span class="math inline">\(\mathbf{y}\)</span> is a linear function of the genotype <span class="math inline">\(\mathbf{G}\)</span> <span class="math display">\[
\mathbf{y} = \mathbf{G}\boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span> <span class="math inline">\(\mathbf{y} \in \mathbb{R}^N\)</span>, <span class="math inline">\(\mathbf{G} \in \mathbb{R}^{N \times P}\)</span>, <span class="math inline">\(\boldsymbol{\beta} \in \mathbb{R}^P\)</span>, and <span class="math inline">\(\boldsymbol{\varepsilon} \in \mathbb{R}^N\)</span></p>
<p><span class="math inline">\(\mathbf{G}\)</span> contains <em>all</em> positions the genome including genotyped ones</p>
<p><span class="math inline">\(N\)</span>: number of samples, <span class="math inline">\(P\)</span>: length of the genome</p>
</div>
</section>
<section id="how-do-we-get-traits" class="slide level2">
<h2>How do we get traits?</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="1">
<p><img data-src="imgs/tree-branch.svg" class="absolute" style="top: 200px; left: 0px; height: 600px; "></p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="2">
<p><img data-src="imgs/tree-0.svg" class="absolute" style="top: 200px; left: 0px; height: 600px; "> <span class="math display">\[
\mathbf{y}_1=\boldsymbol{\beta}_1+\boldsymbol{\beta}_2, \;
\mathbf{y}_2 = \boldsymbol{\beta}_2, \;
\mathbf{y}_3 = \boldsymbol{\beta}_2
\]</span></p>
</div>
<div class="fragment" data-fragment-index="3">
<p><img data-src="imgs/tree-1.svg" class="absolute" style="top: 200px; left: 0px; height: 600px; "> <span class="math display">\[
\mathbf{y}_1=\boldsymbol{\beta}_4, \;
\mathbf{y}_2=\boldsymbol{\beta}_4, \;
\mathbf{y}_3 = \boldsymbol{\beta}_3
\]</span></p>
</div>
</div>
</div><div class="column center" style="width:50%;">
<div class="fragment" data-fragment-index="1">
<p>Consider a local tree that spans over a region</p>
<p>We get trait values by adding up effect sizes (<span class="math inline">\(\boldsymbol{\beta}\)</span>)</p>
</div>
<div class="fragment" data-fragment-index="2">
<ul>
<li><span class="math inline">\(\mathbf{y}_n = \mathbf{G}_{n1} \boldsymbol{\beta}_1 + \mathbf{G}_{n2} \boldsymbol{\beta}_2\)</span></li>
</ul>
</div>
<div class="fragment" data-fragment-index="3">
<ul>
<li><span class="math inline">\(\mathbf{y}_n = \mathbf{G}_{n3} \boldsymbol{\beta}_3 + \mathbf{G}_{n4} \boldsymbol{\beta}_4\)</span></li>
</ul>
</div>
</div></div>
</section>
<section id="branch-centric-view-of-trait-transmission" class="slide level2">
<h2>Branch-centric view of trait transmission</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="r-stack">
<div class="fragment" data-fragment-index="2">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/tree-branch.svg" class="quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="7">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/tree-0.svg" class="quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="8">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/tree-1.svg" class="quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="fragment" data-fragment-index="1">
<p>Inherit a branch first, then a mutation</p>
<div class="fragment" data-fragment-index="3">
<ul>
<li>Sample <span class="math inline">\(1\)</span> inherits edges <span class="math inline">\(1-4\)</span> and <span class="math inline">\(4-5\)</span></li>
</ul>
</div>
<div class="fragment" data-fragment-index="4">
<ul>
<li>Sample <span class="math inline">\(2\)</span> inherits edges <span class="math inline">\(2-4\)</span> and <span class="math inline">\(4-5\)</span></li>
</ul>
</div>
<div class="fragment" data-fragment-index="5">
<ul>
<li>Sample <span class="math inline">\(3\)</span> inherits edge <span class="math inline">\(3-5\)</span></li>
</ul>
</div>
</div>
<div class="fragment" data-fragment-index="6">
<p>Branch’s effect <span class="math inline">\(=\)</span> Sum of mutations’ effect</p>
</div>
<div class="fragment" data-fragment-index="7">
<ul>
<li>Effect of <span class="math inline">\(4-5 = 0\)</span> (1st realization)</li>
</ul>
</div>
<div class="fragment" data-fragment-index="8">
<ul>
<li>Effect of <span class="math inline">\(4-5 = \boldsymbol{\beta}_4\)</span> (2nd realization)</li>
</ul>
</div>
<div class="fragment" data-fragment-index="9">
<p>Branch effect is a random variable!</p>
</div>
</div></div>
</section>
<section id="from-variants-to-branches" class="slide level2 center">
<h2>From variants to branches</h2>
<div class="fragment">
<p><span class="math display">\[
\text{Trait} = \sum_p \text{Variant$_p$ effect size}
\quad
\Rightarrow
\quad
\text{Trait} = \sum_e \text{Branch$_e$ effect size}
\]</span></p>
</div>
<div class="fragment">
<p><span class="math display">\[
\boldsymbol{\upsilon}_e = \text{Branch$_e$ effect size} = \sum_p \text{Variant$_p$ on Branch$_e$}
\]</span></p>
</div>
<div class="fragment">
<p><span class="math display">\[
\mathbf{y} = \sum_p \mathbf{G}_p\boldsymbol{\beta}_p + \boldsymbol{\varepsilon}
\quad
\Rightarrow
\quad
\mathbf{y} = \sum_e \mathbf{Z}_e\boldsymbol{\upsilon}_e + \boldsymbol{\varepsilon}
\]</span> where <span class="math inline">\(\mathbf{Z}_{ne}=\)</span> the number of haplotypes of <span class="math inline">\(n\)</span> that inherit <span class="math inline">\(e\)</span></p>
</div>
</section></section>
<section>
<section id="ancestral-recombination-graph-linear-mixed-model-arg-lmm" class="title-slide slide level1 center">
<h1>Ancestral recombination graph linear mixed model (ARG-LMM)</h1>
<div class="fragment">
<p>Split <span class="math inline">\(\boldsymbol{\upsilon}\)</span> to <span class="math inline">\(\mathbf{u} = \boldsymbol{\upsilon} - \mathrm{E}\boldsymbol{\upsilon}\)</span> and <span class="math inline">\(\mathbf{f}= \mathrm{E}\boldsymbol{\upsilon}\)</span></p>
<p><span class="math display">\[
\mathbf{y} = \underbrace{\mathbf{Z} \mathbf{u}}_{\text{Random effects}} + \underbrace{\mathbf{Z} \mathbf{f}}_{\text{Fixed effects}} + \boldsymbol{\varepsilon}
\]</span></p>
</div>
<div class="fragment">
<p>This is the ancestral recombination graph linear mixed model (ARG-LMM) and <span class="math inline">\(\mathbf{Z} \mathrm{Cov}(\mathbf{u}) \mathbf{Z}^T\)</span> is the expected genetic relatedness matrix (eGRM) <span class="citation" data-cites="Fan2022 Zhang2023">(<a href="#/references" role="doc-biblioref" onclick="">Fan, Mancuso, and Chiang 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Zhang et al. 2023</a>)</span></p>
</div>
<div>
<ul>
<li class="fragment">The random effects are tied to a physical process - Mutations!</li>
<li class="fragment">We start from more lower-level evolutionary statements to recover mixed model assumptions</li>
<li class="fragment">Independent random effects, random effect weights, normality, <span class="math inline">\(\ldots\)</span></li>
</ul>
</div>
</section>
<section id="how-do-we-weigh-branches-of-the-arg" class="slide level2">
<h2>How do we weigh branches of the ARG?</h2>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p><span class="math display">\[
\small l_e:\text{ length in time} \quad s_e:\text{ span in base pairs}
\]</span> <img data-src="imgs/edge_weight-0.svg" class="quarto-figure quarto-figure-center" height="500"></p>
</div>
<div class="fragment">
<p><span class="math display">\[
\small l_e:\text{ length in time} \quad s_e:\text{ span in base pairs}
\]</span> <img data-src="imgs/edge_weight-1.svg" class="quarto-figure quarto-figure-center" height="500"></p>
</div>
</div>
<div class="fragment">
<p><span class="math display">\[
\small \mathrm{Var}(\mathbf{u}_e) \quad \propto \quad \text{Number of mutations} \quad \propto \quad \text{Area}=l_es_e
\]</span></p>
</div>
</section></section>
<section>
<section id="complex-traits-through-the-lens-of-arg-lmm" class="title-slide slide level1 center">
<h1>Complex traits through the lens of ARG-LMM</h1>
<p><span class="math display">\[
\text{What does ARG-LMM tell us about complex trait analysis?}
\]</span></p>
</section>
<section id="genetic-value-covariance" class="slide level2">
<h2>Genetic value covariance</h2>
<div class="r-stack">
<div class="fragment" data-fragment-index="1">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-0.svg" class="quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="4">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-1.svg" class="quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
</div>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="2">
<p><span class="math inline">\(\mathbf{y}_1 = \mathbf{u}_{13} + \mathbf{u}_{34} \quad \text{and} \quad \mathbf{y}_2 = \mathbf{u}_{23} + \mathbf{u}_{34}\)</span></p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="3">
<p><span class="math display">\[
\mathrm{Cov}(\mathbf{y}_1, \mathbf{y}_2) =
\mathrm{Cov}(\mathbf{u}_{13} + \mathbf{u}_{34}, \mathbf{u}_{23} + \mathbf{u}_{34}) =
\mathrm{Cov}(\mathbf{u}_{34}, \mathbf{u}_{34}) =
\mathrm{Var}(\mathbf{u}_{34}) \propto t_3 - t_2
\]</span></p>
</div>
<div class="fragment" data-fragment-index="4">
<p><span class="math display">\[
\mathrm{Cov}(\mathbf{y}_1, \mathbf{y}_2) =
\mathrm{Cov}(\mathbf{u}_{13} + \textcolor{red}{\mathbf{u}_{34}}, \mathbf{u}_{23} + \textcolor{red}{\mathbf{u}_{34}}) =
\mathrm{Cov}(\textcolor{red}{\mathbf{u}_{34}}, \textcolor{red}{\mathbf{u}_{34}}) =
\mathrm{Var}(\textcolor{red}{\mathbf{u}_{34}}) \propto t_3 - t_2
\]</span></p>
</div>
</div>
</section>
<section id="heritability-is-ill-defined-in-arg-lmm" class="slide level2">
<h2>Heritability is <em>ill</em>-defined in ARG-LMM</h2>
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="1">
<p><span class="math display">\[
\small \text{Heritability: }h_g^2 = \frac{\mathrm{Var}(\mathbf{g}_n)}{\mathrm{Var}(\mathbf{y}_n)}
=\frac{\mathrm{Var}(\mathbf{g}_n)}{\mathrm{Var}(\mathbf{g}_n)+\mathrm{Var}(\boldsymbol{\varepsilon}_n)}
\]</span> This applies to all individuals <span class="math inline">\(\small n \in \{1,\ldots,N\}\)</span></p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="2">
<p>However, all individuals have a different amount of genetic variance (except haploids) <span class="math display">\[
\small \mathrm{Var}(\mathbf{g}_n) = \mathrm{Var}(\mathbf{h}_{n1}+\mathbf{h}_{n2}) = \mathrm{Var}(\mathbf{h}_{n1}) + \mathrm{Var}(\mathbf{h}_{n2}) + 2\mathrm{Cov}(\mathbf{h}_{n1},\mathbf{h}_{n2})
\]</span></p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="3">
<p><span class="math display">\[
\small \mathrm{Var}(\mathbf{g}_n) = \mathrm{Var}(\mathbf{h}_{n1}+\mathbf{h}_{n2}) = \mathrm{Var}(\mathbf{h}_{n1}) + \mathrm{Var}(\mathbf{h}_{n2}) + 2\textcolor{red}{\underbrace{\mathrm{Cov}(\mathbf{h}_{n1},\mathbf{h}_{n2})}_{\text{Self-relatedness}}}
\]</span></p>
</div>
<div class="fragment" data-fragment-index="4">
<p>We can’t define a single quantitity <span class="math inline">\(h^2_g=\frac{\textcolor{red}{\mathrm{Var}(\mathbf{g}_n)}}{\textcolor{red}{\mathrm{Var}(\mathbf{g}_n)} + \mathrm{Var}(\boldsymbol{\varepsilon}_n)}\)</span> for everyone</p>
</div>
</div>
<div class="columns">
<div class="column" style="width:50%;">
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="2">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-short-0.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="3">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-short-1.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="2">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-long-0.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="3">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/covariance-long-1.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="polygenic-prediction-is-constrained-by-demography" class="slide level2">
<h2>Polygenic prediction is constrained by demography</h2>
<div class="r-stack">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/geneticvariance.svg" class="fragment quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
<div class="fragment fade-out">
<p><img data-src="imgs/covariance-1.svg" class="absolute" style="top: 70px; right: 50px; height: 550px; "></p>
</div>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p>Some people are less genetically variable than others</p>
</div>
<div class="fragment fade-in-then-out">
<p>Some people are harder to predict genetically than others</p>
</div>
<div class="fragment">
<p>Some populations are <span style="color:red">inherently harder</span> to predict!</p>
</div>
</div>
<div class="footer">
<p>Demographic model from <span class="citation" data-cites="Browning2018">(<a href="#/references" role="doc-biblioref" onclick="">Browning et al. 2018</a>)</span></p>
</div>
</section></section>
<section>
<section id="textsftslmm-fitting-arg-lmm-to-tree-sequences" class="title-slide slide level1 center">
<h1><span class="math inline">\(\textsf{tslmm}\)</span>, fitting ARG-LMM to tree sequences</h1>
<div class="fragment">
<p><span class="math inline">\(\textsf{tslmm}\)</span> utilizes an efficient <em>genetic relatedness matrix - vector product</em> to fit the restricted maximum likelihood (REML) objective</p>
</div>
<div class="fragment">
<p>It can estimate variance components and compute polygenic scores by best linear unbiased prediction (BLUP)</p>
</div>
</section>
<section id="the-matrix-vector-product-algorithm" class="slide level2">
<h2>The matrix-vector product algorithm</h2>
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="1">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/two-trees-empty.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="2">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/two-trees-naive.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
<div class="fragment" data-fragment-index="3">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/two-trees-efficient.svg" class="quarto-figure quarto-figure-center" height="550"></p>
</figure>
</div>
</div>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out" data-fragment-index="1">
<p>The algorithm needs to pass mutations to the correct samples</p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="2">
<p>A naive approach is to push the mutations down to the leaves every time</p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="3">
<p>Wait until the subtree’s topology changes due to edge insertion/deletion</p>
</div>
<div class="fragment fade-in-then-out" data-fragment-index="4">
<p>The wrong recipient will receive the mutations if we procrastinate further</p>
</div>
<div class="fragment" data-fragment-index="5">
<p>Fitting REML <span class="math inline">\(\mathcal{O}(n_s^3) \; \Rightarrow \; \mathcal{O}(n_s+n_t\log n_s)\)</span></p>
<p><span class="math inline">\(n_s\)</span>: number of samples, <span class="math inline">\(n_t\)</span>: number of trees</p>
</div>
</div>
<div class="footer">
<p>Modified figures by Nathaniel Pope (Oregon)</p>
</div>
</section>
<section id="runtime-for-variance-component-estimation" class="slide level2">
<h2>Runtime for variance component estimation</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/runtime.svg" class="fragment quarto-figure quarto-figure-center" height="500"></p>
</figure>
</div>
<div class="fragment">
<p>The runtime scales linearly with respect to the number of individuals (genome length = <span class="math inline">\(10^8\)</span>)</p>
</div>
</section>
<section id="best-linear-unbiased-prediction-blup" class="slide level2">
<h2>Best linear unbiased prediction (BLUP)</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/prediction.svg" class="fragment quarto-figure quarto-figure-center" height="600"></p>
</figure>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p>We measured the accuracy of polygenic scores computed from <span class="math inline">\(\textsf{tslmm}\)</span></p>
</div>
<div class="fragment fade-in-then-out">
<p>Training and testing on two non-overlapping groups embedded in the same tree sequence</p>
</div>
<div class="fragment">
<p>True trees are better, but inferred trees are not too behind!</p>
</div>
</div>
</section>
<section id="summary-future-directions" class="slide level2 center">
<h2>Summary &amp; Future directions</h2>
<p>ARG-LMM lays an explicit connection between population and quantitative genetics</p>
<div class="fragment">
<p>Pseudoreplication due to shared ancestry <span class="citation" data-cites="Rosenberg2009">(<a href="#/references" role="doc-biblioref" onclick="">Rosenberg and VanLiere 2009</a>)</span></p>
</div>
<div class="fragment">
<p>Missing heritability, Mutations vs Mendelian segregation</p>
</div>
<div class="fragment">
<p>A powerful trait simulator based on ARG-LMM <span class="citation" data-cites="Cranmer2020">(<a href="#/references" role="doc-biblioref" onclick="">Cranmer, Brehmer, and Louppe 2020</a>)</span></p>
</div>
<div class="fragment">
<p>Super interesting technical details and proofs (10+ backup slides prepared)</p>
</div>
<div class="fragment">
<p>Predicting polygenic scores of internal nodes <span class="citation" data-cites="Edge2018 Peng2024">(<a href="#/references" role="doc-biblioref" onclick="">Edge and Coop 2018</a>; <a href="#/references" role="doc-biblioref" onclick="">Peng, Mulder, and Edge 2024</a>)</span></p>
</div>
<div class="fragment">
<p>Time conditioned analysis (random vs fixed effects) <span class="citation" data-cites="Fan2022">(<a href="#/references" role="doc-biblioref" onclick="">Fan, Mancuso, and Chiang 2022</a>)</span></p>
</div>
</section></section>
<section>
<section id="thank-you-for-listening" class="title-slide slide level1 center">
<h1>Thank you for listening</h1>
<p align="center">
<img data-src="imgs/qr_code.svg" height="400">
</p>
<p>Link to <span class="citation" data-cites="Lehmann2025">(<a href="#/references" role="doc-biblioref" onclick="">Lehmann et al. 2025</a>)</span>, <span class="math inline">\(\textsf{tslmm}\)</span> preprint coming soon</p>
<p>Collaborators: Nathaniel Pope (Oregon), Jerome Kelleher (Oxford), Gregor Gorjanc (Edinburgh), and Peter Ralph (Oregon)</p>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Browning2018" class="csl-entry" role="listitem">
Browning, Sharon R., Brian L. Browning, Martha L. Daviglus, Ramon A. Durazo-Arvizu, Neil Schneiderman, Robert C. Kaplan, and Cathy C. Laurie. 2018. <span>“Ancestry-Specific Recent Effective Population Size in the Americas.”</span> Edited by Kirk E. Lohmueller. <em>PLOS Genetics</em> 14 (5): e1007385. <a href="https://doi.org/10.1371/journal.pgen.1007385">https://doi.org/10.1371/journal.pgen.1007385</a>.
</div>
<div id="ref-Cranmer2020" class="csl-entry" role="listitem">
Cranmer, Kyle, Johann Brehmer, and Gilles Louppe. 2020. <span>“The Frontier of Simulation-Based Inference.”</span> <em>Proceedings of the National Academy of Sciences</em> 117 (48): 30055–62. <a href="https://doi.org/10.1073/pnas.1912789117">https://doi.org/10.1073/pnas.1912789117</a>.
</div>
<div id="ref-Edge2018" class="csl-entry" role="listitem">
Edge, Michael D, and Graham Coop. 2018. <span>“Reconstructing the History of Polygenic Scores Using Coalescent Trees.”</span> <em>Genetics</em> 211 (1): 235–62. <a href="https://doi.org/10.1534/genetics.118.301687">https://doi.org/10.1534/genetics.118.301687</a>.
</div>
<div id="ref-Fan2022" class="csl-entry" role="listitem">
Fan, Caoqi, Nicholas Mancuso, and Charleston W. K. Chiang. 2022. <span>“A Genealogical Estimate of Genetic Relationships.”</span> <em>The American Journal of Human Genetics</em> 109 (5): 812–24. <a href="https://doi.org/10.1016/j.ajhg.2022.03.016">https://doi.org/10.1016/j.ajhg.2022.03.016</a>.
</div>
<div id="ref-Lehmann2025" class="csl-entry" role="listitem">
Lehmann, Brieuc, Hanbin Lee, Luke Anderson-Trocme, Jerome Kelleher, Gregor Gorjanc, and Peter L. Ralph. 2025. <span>“On ARGs, Pedigrees, and Genetic Relatedness Matrices,”</span> March. <a href="https://doi.org/10.1101/2025.03.03.641310">https://doi.org/10.1101/2025.03.03.641310</a>.
</div>
<div id="ref-Peng2024" class="csl-entry" role="listitem">
Peng, Dandan, Obadiah J. Mulder, and Michael D. Edge. 2024. <span>“Evaluating ARG-Estimation Methods in the Context of Estimating Population-Mean Polygenic Score Histories,”</span> May. <a href="https://doi.org/10.1101/2024.05.24.595829">https://doi.org/10.1101/2024.05.24.595829</a>.
</div>
<div id="ref-Rosenberg2009" class="csl-entry" role="listitem">
Rosenberg, Noah A., and Jenna M. VanLiere. 2009. <span>“Replication of Genetic Associations as Pseudoreplication Due to Shared Genealogy.”</span> <em>Genetic Epidemiology</em> 33 (6): 479–87. <a href="https://doi.org/10.1002/gepi.20400">https://doi.org/10.1002/gepi.20400</a>.
</div>
<div id="ref-SalehiNowbandegani2023" class="csl-entry" role="listitem">
Salehi Nowbandegani, Pouria, Anthony Wilder Wohns, Jenna L. Ballard, Eric S. Lander, Alex Bloemendal, Benjamin M. Neale, and Luke J. O’Connor. 2023. <span>“Extremely Sparse Models of Linkage Disequilibrium in Ancestrally Diverse Association Studies.”</span> <em>Nature Genetics</em> 55 (9): 1494–1502. <a href="https://doi.org/10.1038/s41588-023-01487-8">https://doi.org/10.1038/s41588-023-01487-8</a>.
</div>
<div id="ref-Wakeley2008" class="csl-entry" role="listitem">
Wakeley, John. 2008. <em>Coalescent Theory</em>. Greenwood Village, CO: Roberts &amp; Company.
</div>
<div id="ref-Wong2024" class="csl-entry" role="listitem">
Wong, Yan, Anastasia Ignatieva, Jere Koskela, Gregor Gorjanc, Anthony W Wohns, and Jerome Kelleher. 2024. <span>“A General and Efficient Representation of Ancestral Recombination Graphs.”</span> Edited by G Coop. <em>GENETICS</em> 228 (1). <a href="https://doi.org/10.1093/genetics/iyae100">https://doi.org/10.1093/genetics/iyae100</a>.
</div>
<div id="ref-Zhang2023" class="csl-entry" role="listitem">
Zhang, Brian C., Arjun Biddanda, Árni Freyr Gunnarsson, Fergus Cooper, and Pier Francesco Palamara. 2023. <span>“Biobank-Scale Inference of Ancestral Recombination Graphs Enables Genealogical Analysis of Complex Traits.”</span> <em>Nature Genetics</em> 55 (5): 768–76. <a href="https://doi.org/10.1038/s41588-023-01379-x">https://doi.org/10.1038/s41588-023-01379-x</a>.
</div>
</div>
</section></section>
<section>
<section id="technical-notes" class="title-slide slide level1 center">
<h1>Technical Notes</h1>

</section>
<section id="edge-splitting" class="slide level2">
<h2>Edge splitting</h2>
<div>
<ul>
<li class="fragment">Nodes and edges are reused across multiple trees in a tree sequence</li>
<li class="fragment">Edges, in particular, may not have a unique set of samples along their span</li>
<li class="fragment">Salehi Nowbandegani and colleagues <em>bricked</em> the edges to divide them <span class="citation" data-cites="SalehiNowbandegani2023">(<a href="#/references" role="doc-biblioref" onclick="">Salehi Nowbandegani et al. 2023</a>)</span></li>
<li class="fragment">Henceforth, we assume that edges are splitted to have a unique subtopology <span class="math display">\[
\mathbf{Z}_{ne} = \text{The number of haplotypes of individual $n$ that inherit $e$}
\]</span> The overall matrix <span class="math inline">\(\mathbf{Z}\)</span> is an individual-edge design matrix.</li>
</ul>
</div>
</section>
<section id="collapsing-variants-to-edges" class="slide level2">
<h2>Collapsing variants to edges</h2>
<div>
<ul>
<li class="fragment">When does an individual possess a derived allele? (<span class="math inline">\(\text{ancestral}=0\)</span>, <span class="math inline">\(\text{derived}=1\)</span>)</li>
<li class="fragment">Let <span class="math inline">\(\mathbf{1}_{ep}\)</span> be the indicator <em>random</em> variable of a mutation at edge <span class="math inline">\(e\)</span> and position <span class="math inline">\(p\)</span></li>
</ul>
</div>
<center>
<div class="fragment">
<p>An individual should be a descendant of an edge (<span class="math inline">\(\mathbf{Z}_{ne}=1\)</span>)</p>
</div>
<div class="fragment">
<p><span class="math display">\[
+
\]</span> That edge should have mutation (<span class="math inline">\(\mathbf{1}_{ep}=1\)</span>)</p>
</div>
</center>
<div class="fragment">
<p><span class="math display">\[
\mathbf{G}_{np} = \sum_{e: p \in e} \mathbf{Z}_{ne} \mathbf{1}_{ep} \quad \Leftrightarrow \quad \mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}
\]</span></p>
</div>
<div>
<ul>
<li class="fragment">Assumes that there are no parent-child mutation pairs, but allows <em>some</em> recurrent mutations</li>
</ul>
</div>
</section>
<section id="exchange-the-summations" class="slide level2">
<h2>Exchange the summations</h2>
<div class="fragment">
<p>Recall <span class="math inline">\(\mathbf{G}_p = \sum_{e:p \in e} \mathbf{Z}_{e} \mathbf{1}_{ep}\)</span> and <span class="math inline">\(\mathbf{y} = \sum_{p=1}^P \mathbf{G}_p \boldsymbol{\beta}_p + \boldsymbol{\varepsilon}\)</span></p>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p>Substitute <span class="math inline">\(\mathbf{G}_p\)</span> <span class="math display">\[
\textcolor{red}{\sum_{p=1}^P \sum_{e:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon}
\]</span></p>
</div>
<div class="fragment fade-in-then-out">
<p>Exchange the inner and the outer summation <span class="math display">\[
\textcolor{red}{\sum_{e=1}^E \sum_{p:p \in e}} \mathbf{Z}_e \boldsymbol{\beta}_p \mathbf{1}_{ep} + \boldsymbol{\varepsilon}
\]</span></p>
</div>
<div class="fragment">
<p>Pull out <span class="math inline">\(\mathbf{Z}_e\)</span> and group the positions nested in <span class="math inline">\(p: p \in e\)</span> <span class="math display">\[
\begin{aligned}
    &amp; \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\left(\sum_{p:p\in e} \boldsymbol{\beta}_p \mathbf{1}_{ep} \right)} + \boldsymbol{\varepsilon} \\
    &amp;= \sum_{e=1}^E \mathbf{Z}_e \textcolor{blue}{\boldsymbol{\upsilon}_e} + \boldsymbol{\varepsilon} \\
    &amp;= \mathbf{Z} \boldsymbol{\upsilon} + \boldsymbol{\varepsilon}
\end{aligned}
\]</span></p>
</div>
</div>
<div class="fragment">
<p><span class="math inline">\(\boldsymbol{\upsilon}\)</span> is a random variable made up of mutation-driven random variables <span class="math inline">\(\mathbf{1}_{ep}\)</span>!</p>
</div>
</section>
<section id="random-effects-are-independent" class="slide level2">
<h2>Random effects are independent</h2>
<div>
<ul>
<li class="fragment"><p>Independent entries of random effects is a key assumption of linear mixed models</p></li>
<li class="fragment"><p>This can be <em>proved</em> in ARG-LMM, instead of assuming it <span class="math display">\[
  \mathrm{Cov}( \mathbf{u}_e, \mathbf{u}_{e'} ) = \sum_{p \in e,e'} \boldsymbol{\beta}_p^2 \mathrm{Cov}(\mathbf{1}_{ep}, \mathbf{1}_{e'p})
\]</span></p></li>
<li class="fragment"><p>The covariance between the indicators are higher-order terms of mutation rates, so we ignore it <span class="citation" data-cites="Wakeley2008">(<a href="#/references" role="doc-biblioref" onclick="">Wakeley 2008</a>)</span> <span class="math display">\[
  \begin{aligned}
  \mathrm{Cov}(\mathbf{1}_{ep}, \mathbf{1}_{e'p}) &amp;= \mathrm{E}[\mathbf{1}_{ep}\mathbf{1}_{e'p}] - \mathrm{E}[\mathbf{1}_{e'p}]\mathrm{E}[\mathbf{1}_{ep}] \\
      &amp;= 0 -l_eu_{ep}l_{e'}u_{e'p} \approx 0
  \end{aligned}
\]</span> where <span class="math inline">\(l_e\)</span> is the (time-)length of edge <span class="math inline">\(e\)</span>.</p></li>
</ul>
</div>
</section>
<section id="the-marginal-distribution-of-mathbfu_e" class="slide level2">
<h2>The marginal distribution of <span class="math inline">\(\mathbf{u}_e\)</span>?</h2>
<div>
<ul>
<li class="fragment"><p>The Gaussian prior on random effects is a popular choice</p></li>
<li class="fragment"><p>One might be tempted to invoke the central limit theorem to <span class="math inline">\(\mathbf{u}_e\)</span> (sum of indicators) <span class="math display">\[
      \mathbf{u}_e \bigg/ \sqrt{l_e s_e} \cdot \sqrt{ \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep} }
      \rightarrow N(0,1^2) \text{ as } s_e \rightarrow \infty
  \]</span> where <span class="math inline">\(s_e\)</span> is the span (in base pairs) of edge <span class="math inline">\(e\)</span></p></li>
<li class="fragment"><p>The convergence is unlikely to be fast enough given the small value of <span class="math inline">\(\mathbf{E}[\mathbf{1}_{ep}]\)</span> (Berry-Esseen).</p></li>
<li class="fragment"><p>Fortunately, the variance is computable and is <span class="math display">\[
      \mathrm{Var}(\mathbf{u}_e) = l_e s_e \cdot \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep}
  \]</span></p></li>
</ul>
</div>
</section>
<section id="more-on-mathrmvarmathbfu_e" class="slide level2">
<h2>More on <span class="math inline">\(\mathrm{Var}(\mathbf{u}_e)\)</span></h2>
<div>
<ul>
<li class="fragment">The weight <span class="math inline">\(\mathrm{Var}(\mathbf{u}_e)\)</span> has two components</li>
<li class="fragment">The area <span class="math inline">\(l_e s_e\)</span></li>
<li class="fragment">Mutation rate-weighted squared average of effect sizes <span class="math display">\[
      \tau_e^2 = \frac{1}{s_e} \sum_{p:p \in e} \boldsymbol{\beta}_p^2 u_{ep}
  \]</span></li>
<li class="fragment">As a measure of functional significance, variance components are confounded by the area</li>
</ul>
</div>
</section>
<section id="fixed-effects-are-constant-under-neutrality" class="slide level2">
<h2>Fixed effects are constant under neutrality</h2>
<div>
<ul>
<li class="fragment">Suppose that <span class="math inline">\(u_{ep}=u_p\)</span>, i.e., the mutation rate is constant across edges for a given position</li>
</ul>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p><span class="math display">\[
\begin{aligned}
    &amp;\left[ \mathbf{Z} \mathbf{f} \right]_n = \sum_{e=1}^E \mathbf{Z}_{ne} \mathbf{E}
    \left[ \sum_{p: p \in e} \boldsymbol{\beta}_p\mathbf{1}_{ep} \right] \\
    \end{aligned}
\]</span></p>
</div>
<div class="fragment">
<p><span class="math display">\[
\sum_{p=1}^P \boldsymbol{\beta}_p u_p
    \left(
            \sum_{e: p \in e} \mathbf{Z}_{ne} l_e
    \right)
    = \sum_{p=1}^P \boldsymbol{\beta}u_p \cdot 2t_{\mathrm{root},p} = \text{const. resp. to $n$}
\]</span></p>
</div>
</div>
<div>
<ul>
<li class="fragment"><p>An intercept is enough to account for the fixed effects <span class="math inline">\(\mathbf{Z}\mathbf{f}\)</span> under this condition</p></li>
<li class="fragment"><p>The assumption is standard in neutral settings</p></li>
</ul>
</div>
<div class="fragment">
<p><span class="math display">\[
    \text{\textcolor{red}{Conjecture:} selection $\Rightarrow$ fixed effects?}
\]</span></p>
</div>
</section>
<section id="becareful-of-pseudoreplication" class="slide level2">
<h2>Becareful of pseudoreplication</h2>
<div>
<ul>
<li class="fragment">Non-overlapping samples are not independent</li>
<li class="fragment">Everyone shares some amount of mutational history</li>
<li class="fragment">Parameter estimates (e.g.&nbsp;variance components) are correlated</li>
</ul>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="imgs/pseudoreplication.svg" class="fragment quarto-figure quarto-figure-center" height="400"></p>
</figure>
</div>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p>This is also the very reason why BLUP works</p>
</div>
<div class="fragment">
<p>We are all correlated!</p>
</div>
</div>
</section>
<section id="missing-heritability" class="slide level2">
<h2><em>Missing</em> heritability?</h2>
<div class="fragment">
<p>ARG-LMM variance component only reflects mutational variability</p>
</div>
<div>
<ul>
<li class="fragment">Pedigree-based heritability captures Mendelian segregation and mutation is ignored</li>
<li class="fragment">ARG-LMM’s generative model only has mutation and no segregation</li>
<li class="fragment">Why compare quantities stemming from different random forces? <span class="citation" data-cites="Zhang2023">(<a href="#/references" role="doc-biblioref" onclick="">Zhang et al. 2023</a>)</span></li>
</ul>
</div>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="https://whalebonemag.com/wp-content/uploads/2022/11/applesandoranges-lilguys-02.jpg" class="fragment quarto-figure quarto-figure-center" height="400"></p>
</figure>
</div>
<div class="footer">
<p>Figure from <a href="https://whalebonemag.com/">Whalebone Magazine</a></p>
</div>
</section>
<section id="estimation-quality-of-variance-components" class="slide level2">
<h2>Estimation quality of variance components</h2>
<div class="r-stack">
<div class="fragment fade-in-then-out">
<p><strong>Simulated trees</strong></p>
<p><img data-src="imgs/estimation_true.svg" height="700"></p>
</div>
<div class="fragment">
<p><strong>Inferred (tsinfer+tsdate) trees</strong></p>
<p><img data-src="imgs/estimation_infer.svg" height="700"></p>
</div>
</div>
</section>
<section id="there-are-many-genetic-variances" class="slide level2">
<h2>There are many genetic variances</h2>
<div>
<ul>
<li class="fragment">ARG-conditioned variance <span class="math display">\[
\mathrm{Var}(\mathbf{y} \mid \mathrm{ARG})
\]</span></li>
<li class="fragment">Pedigree-conditioned variance <span class="math display">\[
\mathrm{Var}(\mathbf{y} \mid \mathrm{Pedigree})
\]</span></li>
<li class="fragment">Demography-conditioned variance <span class="math display">\[
\mathrm{Var}(\mathbf{y} \mid \mathrm{Demography})
\]</span></li>
<li class="fragment">Time conditioning (reference population)</li>
</ul>
</div>

</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'default',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'slide',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1600,

        height: 900,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>